src/hostapi/wasapi/pa_win_wasapi.c

```diff
@@ -2974,6 +2982,18 @@ static PaError GetClosestFormat(IAudioClient *client, double sampleRate, const P
 
     if (hr == S_OK)
     {
+        // Workaround for the Realtek driver bug (see https://github.com/PortAudio/portaudio/issues/875):
+        // IAudioClient_IsFormatSupported() accepts Mono format (mistakenly) but fails in
+        // IAudioClient_Initialize(), this bug was probably fixed because it can not be reproduced
+        // with Realtek driver version 10.0.19041.3636, 19/10/2023, affects Output mode only
+        if (output && isSharedMode && (params.channelCount == 1) && (sharedClosestMatch == NULL) &&
+            (deviceInfo->MixFormat.Format.nChannels > 1))
+        {
+            // force internal Mono to Stereo mixer
+            outWavex->Format.nChannels = 2;
```

Comment:
What effect does this line have? Does PortAudio then need to duplicate the mono stream? or does this line simply configure the mixer without changing the format of data sent to WASAPI?

Reply:
This line activates internal PA WASAPI's Mono to Stereo mixer (works as duplication of Mono stream to L and R channels, no data alteration). So WASAPI backend receives Stereo stream, user-side code sends Mono and this operation is transparent for user-code as it just keeps sending Mono data as it was meant by the code. Normally WASAPI backend would just report that 1 channel is not supported and provide 2 channel format in return but this buggy driver (as I understood it is quite widespread, even nowadays) does not do it. So we enforce 2 channels output for it.